Recrutify/Recrutify.Services/Services/ScheduleService.cs

```diff
@@ -81,6 +93,81 @@ public async Task UpdateScheduleSlotsForCurrentUserAsync(IEnumerable<DateTime> d
             await _scheduleRepository.BulkUpdateScheduleSlotsAsync(currentUserId, newListDateTime, removedListDateTime);
         }
 
+        public async Task ProcessingAppointOrCancelInterviewsAsync(IEnumerable<InterviewDTO> interviews, Guid projectId, CancellationToken cancellationToken)
+        {
+            var projectExists = await _projectService.ExistsAsync(projectId, cancellationToken);
+            if (!projectExists)
+            {
+                throw new Exception("Project does not exist.");
+            }
+
+            var allCadidates = await _candidateService.GetCandidatesByIdsAsync(interviews.Select(i => i.CandidateId));
+
+            var cancelledInterviews = _mapper.Map<IEnumerable<Interview>>(interviews.Where(i => i.IsAppointment == false));
```

Comment:
try to create dictionary with the key of bool type, then it wouldn't be necessary to do Where twice for true and false

Reply:
+